experimental/trusted_shuffler/trusted_shuffler/src/lib.rs

```diff
@@ -22,48 +22,69 @@ mod tests;
 use anyhow::{anyhow, Context};
 use async_trait::async_trait;
 use futures::{stream::FuturesOrdered, StreamExt};
+use hyper::Uri;
 use std::{
-    marker::{Send, Sync},
+    cmp::Ordering,
     mem::take,
     ops::DerefMut,
     sync::{Arc, Mutex},
+    vec,
 };
 use tokio::{sync::oneshot, time::Duration};
 
-type Request = Vec<u8>;
-type Response = Vec<u8>;
+#[derive(Debug, PartialEq)]
+pub struct TrustedShufflerRequest {
+    pub body: Vec<u8>,
+    pub uri: Uri,
+}
+#[derive(Debug, PartialEq)]
+pub struct TrustedShufflerResponse {
+    pub body: Vec<u8>,
+}
+
+impl TrustedShufflerResponse {
+    fn empty() -> TrustedShufflerResponse {
+        TrustedShufflerResponse { body: vec![] }
+    }
+}
 
-// The Trusted Shuffler responds with the `EMPTY_RESPONSE` if the backend does not respond within a
-// given timeout.
-const EMPTY_RESPONSE: Vec<u8> = vec![];
+// Workaround so we can have either a request or a response in `data` of `Message`.
+// We should be able to refactor so we do not need the `Wrapper` (and `panics`) any more.
+#[derive(Debug)]
+enum Wrapper {
+    Request(TrustedShufflerRequest),
+    Response(TrustedShufflerResponse),
+}
+
+// We only compare the body, but we need revisit this and decide if this
+// is how we want to sort our requests and responses.
+fn cmp(w1: &Wrapper, w2: &Wrapper) -> Ordering {
+    match (w1, w2) {
+        (Wrapper::Request(r1), Wrapper::Request(r2)) => r1.body.cmp(&r2.body),
+        (Wrapper::Response(r1), Wrapper::Response(r2)) => r1.body.cmp(&r2.body),
+        (_, _) => panic!(
```

Comment:
Should it panic or is there a special constant in `Ordering` for this?

Reply:
You could also look into implementing the following traits: https://doc.rust-lang.org/std/cmp/index.html

Reply:
Yes, I agree, I'd revisit that once we know how we want to sort our TrustedShufflerRequests/Responses.